docroot/modules/custom/mass_entityreference/src/Plugin/EntityReferenceSelection/MassFilterEntitiesSelection.php

```diff
@@ -31,7 +31,7 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $types = unserialize($cookie);
     }
 
-    $query = $this->entityTypeManager->getStorage($target_type)->getQuery();
+    $query = $this->entityTypeManager->getStorage($target_type)->getQuery()->accessCheck();
```

Comment:
Usually we want accessCheck(FALSE). Is that not the case here?

Reply:
I was aiming for least harm to address a test error. My understanding is that prior to not having to explicitly call accessCheck, this query would have behaved like accessCheck(TRUE).
 
In this case, I think the main concern would be whether authors can view unpublished content they're supposed to see in reference fields. I tested out a reference field again, and can confirm that users who can view unpublished content can still see unpublished content with this check set to true. But, I also noticed when reviewing that we add an entity access tag [here](https://github.com/massgov/openmass/pull/2217/files#diff-0c09f380855341827536faa0a0839d59137f730de4b5d00062b706a39d2519a2R55). We probably don't need both access checks, so happy to change this if it is more likely to be expected.